src/pages/Admin/MapPreview/MapPreview.tsx

```diff
@@ -1,62 +1,83 @@
-import React, { useRef, useState } from "react";
-
-import { MapPreviewProps } from "./MapPreview.types";
-
-import * as S from "./MapPreview.styles";
-import BackgroundSelect from "../BackgroundSelect";
+import React, { useMemo, useRef, useState } from "react";
+import { isEqual } from "lodash";
+import { DndProvider } from "react-dnd";
+import { HTML5Backend } from "react-dnd-html5-backend";
 
 import { updateRoom, updateVenue_v2 } from "api/admin";
+
 import { useUser } from "hooks/useUser";
-import { DndProvider } from "react-dnd";
-import { HTML5Backend } from "react-dnd-html5-backend";
+
 import {
   Container,
   SubVenueIconMap,
 } from "pages/Account/Venue/VenueMapEdition/Container";
-import { isEqual } from "lodash";
 import Legend from "components/atoms/Legend";
+import * as S from "./MapPreview.styles";
+import BackgroundSelect from "../BackgroundSelect";
+import { MapPreviewProps } from "./MapPreview.types";
 
 const MapPreview: React.FC<MapPreviewProps> = ({
   venueName,
   mapBackground,
   rooms,
   venueId,
+  editing,
+  setEditing,
 }) => {
   const { user } = useUser();
-  const [editing, setEditing] = useState<boolean>(false);
+  const [mapRooms, setMapRooms] = useState(rooms);
 
   const roomRef = useRef<SubVenueIconMap>({});
 
-  if (!user) return <></>;
+  const iconsMap = useMemo(
+    () =>
+      mapRooms!.map((room, index: number) => ({
```

Comment:
Is it possible to remove `!` to ensure type safety across the app?

Reply:
Agreed. This is something that comes up relatively often. It would be good to see if there is a way to eslint ban it.

Reply:
Backlogged this general concept in https://github.com/sparkletown/internal-sparkle-issues/issues/149